universal-login-sdk/test/unit/core/AggregateBalanceObserver.ts

```diff
@@ -1,48 +1,82 @@
-import {createMockProvider} from 'ethereum-waffle';
-import {TokenDetailsWithBalance, BalanceChecker, ETHER_NATIVE_TOKEN, TEST_CONTRACT_ADDRESS, TEST_ACCOUNT_ADDRESS} from '@universal-login/commons';
+import {TokenDetailsWithBalance, ETHER_NATIVE_TOKEN, TEST_CONTRACT_ADDRESS} from '@universal-login/commons';
 import {AggregateBalanceObserver} from '../../../lib/core/observers/AggregateBalanceObserver';
 import {BalanceObserver} from '../../../lib/core/observers/BalanceObserver';
-import {PriceOracle} from '../../../lib/core/services/PriceOracle';
+import {PriceObserver} from '../../../lib/core/observers/PriceObserver';
 
 describe('UNIT: AggregateBalanceObserver', () => {
-  const priceOracle = new PriceOracle();
-  const provider = createMockProvider();
-  const balanceChecker = new BalanceChecker(provider);
-  const balanceObserver = new BalanceObserver(balanceChecker, TEST_ACCOUNT_ADDRESS, []);
-  const aggregateBalanceObserver = new AggregateBalanceObserver(balanceObserver, priceOracle);
+  const mockedAggregateBalanceObserver = new AggregateBalanceObserver({} as BalanceObserver, {} as PriceObserver);
+  mockedAggregateBalanceObserver.notifyListeners = () => {};
+
   const tokenDetailsWithBalance: TokenDetailsWithBalance[] = [
     {...ETHER_NATIVE_TOKEN, balance: utils.parseEther('1')},
     {address: TEST_CONTRACT_ADDRESS, symbol: 'Mock', name: 'MockToken', balance: utils.parseEther('2')}
   ];
-  const ETH_PRICE_IN_USD = 1405;
+  const tokenPrices = {
+    ETH: {USD: 218.21, EUR: 194.38, BTC: 0.01893},
+    Mock: {USD: 0.02619, EUR: 0.02337, BTC: 0.00000227}
+  };
+
+  beforeEach(() => {
+    mockedAggregateBalanceObserver.priceObserverCallback(tokenPrices);
+    mockedAggregateBalanceObserver.balanceObserverCallback(tokenDetailsWithBalance);
+  });
+
+  context('getTokenTotalWorth', () => {
+    it('addBalances {} + {USD, EUR, BTC}', () => {
+      const token1TotalWorth = {};
+      const token2TotalWorth = {USD: 1838.51, EUR: 1494.71, BTC: 0.09893};
+
+      const tokensTotalWorth = mockedAggregateBalanceObserver.addBalances(token1TotalWorth, token2TotalWorth);
+
+      expect(tokensTotalWorth).to.deep.equal({USD: 1838.51, EUR: 1494.71, BTC: 0.09893});
+    });
+
+    it('addBalances {USD, EUR, BTC} + {USD, EUR, BTC}', () => {
+      const token1TotalWorth = {USD: 218.21, EUR: 194.38, BTC: 0.01893};
+      const token2TotalWorth = {USD: 1838.51, EUR: 1494.71, BTC: 0.09893};
 
-  context('token balance in USD', async () => {
-    it('ETH', async () => {
-      const ethBalanceInUSD = await aggregateBalanceObserver.getTokenBalance(tokenDetailsWithBalance[0], 'USD');
+      const tokensTotalWorth = mockedAggregateBalanceObserver.addBalances(token1TotalWorth, token2TotalWorth);
 
-      expect(ethBalanceInUSD).to.be.greaterThan(1 * ETH_PRICE_IN_USD - 0.1).and.to.be.lessThan(1 * ETH_PRICE_IN_USD + 0.1);
+      expect(tokensTotalWorth).to.deep.equal({USD: 218.21 + 1838.51, EUR: 194.38 + 1494.71, BTC: 0.01893 + 0.09893});
```

Comment:
In tests, it is much easier to use simple values like 1, 2, 10, 20 and build simple examples.
Makes it more enjoyable to write and easier to read.

```ts
const token1TotalWorth = {USD: 100, EUR: 150, BTC: 0.01};
const token2TotalWorth = {USD: 1000, EUR: 1500, BTC: 0.1};
expect(tokensTotalWorth).to.deep.equal({USD: 1100, EUR: 1659, BTC: 0.11});
```